docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+                rx.table.column_header_cell("Gender"),
+            ),
+        ),
+        rx.table.body(
+            rx.foreach(
+                State.users, show_user
+            ),
+        ),
+        variant="surface",
+        size="3",
+)
+```
+
+
+Next let's add a form to the app so we can add new users to the table.
+
+
+## Using a Form to Add Data 
+
+We build a form using `rx.form`, whick takes several components such as `rx.input` and `rx.select`, which represent the form fields that allow you to add information to submit with the form. Check out here for other form components !!!!!
+
+The `rx.input` component takes in several props. The `placeholder` prop is the text that is displayed in the input field when it is empty. The `name` prop is the name of the input field, which gets passed through in the dictionary when the form is submitted. The `required` prop is a boolean that determines if the input field is required.
+
+The `rx.select` component takes in a list of options that are displayed in the dropdown. The other props used here are identical to the `rx.input` component.
+
+```python demo
+rx.form(
+    rx.input(
+        placeholder="User Name", name="name", required=True
+    ),
+    rx.input(
+        placeholder="user@reflex.dev",
+        name="email",
+    ),
+    rx.select(
+        ["Male", "Female"],
+        placeholder="Male",
+        name="gender",
+    ),
+)
+```
+
+This form is all very compact as you can see from the example, so we need to add some styling to make it look better. We can do this by adding a `vstack` component around the form fields. The `vstack` component stacks the form fields vertically. Check out the [layout] docs !!!! for more information on how to layout your app.
+
+
+```python demo
+rx.form(
+    rx.vstack(
+        rx.input(
+            placeholder="User Name", name="name", required=True
+        ),
+        rx.input(
+            placeholder="user@reflex.dev",
+            name="email",
+        ),
+        rx.select(
+            ["Male", "Female"],
+            placeholder="Male",
+            name="gender",
+        ),
+    ),
+)
+```
+
+Now you have probably realised that we have all the form fields, but we have no way to submit the form. We can add a submit button to the form by adding a `rx.button` component to the `vstack` component. The `rx.button` component takes in the text that is displayed on the button and the `type` prop which is the type of button. The `type` prop is set to `submit` so that the form is submitted when the button is clicked. 
+
+In addition to this we need a way to update the `users` state variable when the form is submitted. All state changes are handled through functions in the state class, called [event handlers]({docs.events.events_overview.path}).
+
+Components have special props called event triggers, such as `on_submit`, that can be used to make components interactive. Event triggers connect components to event handlers, which update the state.
```

Comment:
The way this is described for me is... eh. What does on_submit take in a dictionary? Are there events? What do they take in? Again, maybe I ask too many questions as a new user haha

Reply:
fixed